pkg/operator/configobservation/apiserver/observer_apiserver.go

```diff
@@ -0,0 +1,199 @@
+
+	previouslyObservedConfig := map[string]interface{}{}
+	if currentNamedCertificates, _, err := unstructured.NestedSlice(existingConfig, namedCertificatesPath...); len(currentNamedCertificates) > 0 || err != nil {
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if err := unstructured.SetNestedSlice(previouslyObservedConfig, currentNamedCertificates, namedCertificatesPath...); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	namedCertificates := apiServer.Spec.ServingCerts.NamedCertificates
+	if len(namedCertificates) > 10 {
+		err := fmt.Errorf("apiservers.config.openshift.io/cluster: spec.servingCerts.namedCertificates cannot have more than 10 entries")
+		recorder.Warningf("ObserveNamedCertificatesFailed", err.Error())
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	secretsMap := map[string]string{}
+	if len(namedCertificates) > 0 {
+		var observedNamedCertificates []interface{}
+		for index, namedCertificate := range namedCertificates {
+			observedNamedCertificate := map[string]interface{}{}
+			if len(namedCertificate.Names) > 0 {
+				if err := unstructured.SetNestedStringSlice(observedNamedCertificate, namedCertificate.Names, "names"); err != nil {
+					return previouslyObservedConfig, append(errs, err)
+				}
+			}
+			sourceSecretName := namedCertificate.ServingCertificate.Name
+			if len(sourceSecretName) == 0 {
+				err := fmt.Errorf("apiservers.config.openshift.io/cluster: spec.servingCerts.namedCertificates[%v].servingCertificate.name not found", index)
+				recorder.Warningf("ObserveNamedCertificatesFailed", err.Error())
+				return previouslyObservedConfig, append(errs, err)
+			}
+			targetSecretName := fmt.Sprintf("user-serving-cert-%03d", index)
+			secretsMap[targetSecretName] = sourceSecretName
+			certFile := fmt.Sprintf("/etc/kubernetes/static-pod-resources/secrets/%v/tls.crt", targetSecretName)
+			if err := unstructured.SetNestedField(observedNamedCertificate, certFile, "certFile"); err != nil {
+				return previouslyObservedConfig, append(errs, err)
+			}
+			keyFile := fmt.Sprintf("/etc/kubernetes/static-pod-resources/secrets/%v/tls.key", targetSecretName)
+			if err := unstructured.SetNestedField(observedNamedCertificate, keyFile, "keyFile"); err != nil {
+				return previouslyObservedConfig, append(errs, err)
+			}
+			observedNamedCertificates = append(observedNamedCertificates, observedNamedCertificate)
+		}
+		if err := unstructured.SetNestedField(observedConfig, observedNamedCertificates, namedCertificatesPath...); err != nil {
+			return previouslyObservedConfig, append(errs, err)
+		}
+	}
+
+	for targetSecretName, sourceSecretName := range secretsMap {
```

Comment:
I don't see where you're clearing `SyncSecret` on cases where I had 5 items and went to 4.  Please add a test.

Also a test where I reorder those 4.